recipes-apps/ovmenu-ng-skripts/files/download-igc.sh

```diff
@@ -1,16 +1,39 @@
-#!/bin/bash
-
-USB_PATH="/usb/usbstick/openvario/igc"
-IGC_PATH="/home/root/.xcsoar/logs"
-mkdir -p $USB_PATH
-
-if [ -z "$(ls $IGC_PATH/*.igc 2>/dev/null)" ]; then
-        echo "No files found !!!"
-else
-        for igcfile in $(find $IGC_PATH -name '*.igc'); do
-                echo $igcfile
-                cp $igcfile $USB_PATH
-        done
-fi
-
-umount /dev/sda1
+\#!/bin/bash
+
+USB_PATH="/usb/usbstick/openvario/igc/"
+IGC_PATH="/home/root/.xcsoar/logs/"
+
+filter()
+{
+    NUM=0
+    ls $IGC_PATH -r | egrep '.igc' | while read i ; do
+        echo $'\t'$NUM$'\t'$i
+        NUM=$(($NUM+1))
+    done
+}
+
+LINES=$(filter)
+
+IFS=$'\n\t'
+
+#if [$NUM -gt 0]
+#  then
+    dialog --backtitle "Openvario" --title "Download IGC" --menu "Move using [UP] [DOWN],[Enter] to Select" 30 50 28 $LINES 2>/tmp/menuitem.tmp
+
+#else
+#    dialog --backtitle "Openvario" --title "Download IGC" --msgbox "No IGC files found" 6 20
+#fi
+
+unset IFS
+
+ITEM=$(cat /tmp/menuitem.tmp)
+FILE=$(echo "$LINES" | sed -n 's/\t'$ITEM'\t//p')
+
+#echo $FILE
+
+cp $IGC_PATH/$FILE $USB_PATH/$FILE
+umount	/dev/sda1
+dialog --backtitle "Openvario" --title "Download IGC" --msgbox "Done!!" 6 20
```

Comment:
This "Done" message in the end has one bad consequence: it hides errors from the copy commands. E.g. if your flash drive is not mounted, no files are copied, but you see the "Done" message, that can mislead you into thinking that you've got your igc file.

I think it would be better to just echo "Done" and preserve the log. Make sure you test failure casees too.

Reply:
You are right. Saw this after testing.